### PR TITLE
Use jupyter-repo2docker rather than s2i directly

### LINKED ISSUE #75: Replace s2i with repo2docker
We should use repo2docker in the documentation for jhub!

### PR DESCRIPTION
**Pull request by yuvipanda**
Fixes #75

### CODE REVIEW

File: doc/source/extending-jupyterhub.rst [hunk 1]
```diff
@@ -91,33 +91,39 @@ consumes a lot of memory).
...
 
-s2i, also known as `Source to Image <https://github.com/openshift/source-to-image>`_,
+`repo2docker <https://github.com/jupyter/repo2docker>`_
 lets you quickly convert a GitHub repository into a Docker image that we can use
 as a base for your JupyterHub instance. Anything inside the GitHub repository
 will exist in a user’s environment when they join your JupyterHub. If you
 include a ``requirements.txt`` file in the root level of your of the repository,
-s2i will ``pip install`` each of these packages into the Docker image to be
-built. Below we’ll cover how to use s2i to generate a Docker image and how to
+repo2docker will ``pip install`` each of these packages into the Docker image to be
```
Comment (choldgraf) [inline doc/source/extending-jupyterhub.rst]: this makes it sounds python-specific (which it mostly is right now but not in the future). We should make sure to remember to change this language when we have more fully-featured R/Julia support

File: doc/source/extending-jupyterhub.rst [hunk 2]
```diff
@@ -91,33 +91,39 @@ consumes a lot of memory).
...
+Below we’ll cover how to use repo2docker to generate a Docker image and how to
 configure JupyterHub to build off of this image.
 
-.. note::
-       For this section, you’ll need to install s2i and docker.
+1. **Download and start Docker.** You can do this by downloading and installing
+   Docker at `this link <https://store.docker.com/search?offering=community&platform=desktop%2Cserver&q=&type=edition>`_.
+   Once you’ve started Docker, it will show up as a tiny background application.
 
+2. **Install repo2docker**. You can easily do this with ``pip`` inside.
```
Comment (choldgraf) [inline doc/source/extending-jupyterhub.rst]: ~~inside~~

File: doc/source/extending-jupyterhub.rst [hunk 3]
```diff
@@ -91,33 +91,39 @@ consumes a lot of memory).
...
-   links below:
+   .. code:: bash
 
-       - On OSX: ``brew install s2i``
-       - On Linux and Windows: `follow these instructions
-         <https://github.com/openshift/source-to-image#installation>`_
+      pip install jupyter-repo2docker
+
+
+   If that doesn't work due to permissions, try:
```
Comment (choldgraf) [inline doc/source/extending-jupyterhub.rst]: I feel like these should be embedded in a `.. note::` block

File: doc/source/extending-jupyterhub.rst [hunk 4]
```diff
@@ -132,15 +138,13 @@ configure JupyterHub to build off of this image.
...
+4. **Use repo2docker to build your Docker image.**
+   .. code-block:: bash
 
-       s2i build --exclude "" <git-repo-url>  jupyterhub/singleuser-builder-venv-3.5:v0.1.5 gcr.io/<project-name>/<name-of-image>:<tag>
+      jupyter-repo2docker <YOUR-GITHUB-REPOSITORY> --image=gcr.io/<PROJECT-NAME>/<IMAGE-NAME>:<TAG> --no-run
 
-   this effectively says *s2i, build `<this repository>` to a Docker image by
-   using `<this template>` and call the image `<this>`*. The `--exclude ""` ensures
-   that all files are included in the container (e.g. `.git` directory).
+   This tells repo2docker to fetch master of the github repository, and use
```
Comment (choldgraf) [inline doc/source/extending-jupyterhub.rst]: should we document what `--no-run` means?

File: doc/source/extending-jupyterhub.rst (hunk 1 above)
Comment (yuvipanda) [inline doc/source/extending-jupyterhub.rst]: +1. We also need actual docs for repo2docker to link to from here.

**Pull request merged**